src/Phobos.Kafka/src/Petabridge.Phobos.Kafka.Producer/AkkaService.cs

```diff
@@ -95,12 +107,40 @@ public RouterForwarderActor(IActorRef routerActor)
     /// </summary>
     public sealed class AkkaActors
     {
+        public const string KafkaServiceHost = "KAFKA_SERVICE_HOST";
+        public const string KafkaServicePort = "KAFKA_SERVICE_PORT";
+        
+        private const string Topic = "demo";
+
         public AkkaActors(ActorSystem sys)
         {
             Sys = sys;
-            ProducerActor = sys.ActorOf(Props.Create(() => new ProducerActor()), "kafka-producer");
-            ConsoleActor = sys.ActorOf(Props.Create(() => new ConsoleActor(ProducerActor)), "console");
             RouterActor = sys.ActorOf(Props.Empty.WithRouter(FromConfig.Instance), "echo");
+            
+            var log = Logging.GetLogger(sys, "KafkaProducerStream");
+            
+            var kafkaHost = Environment.GetEnvironmentVariable(KafkaServiceHost);
+            var kafkaPort = Environment.GetEnvironmentVariable(KafkaServicePort);
+            var bootstrapServer = $"{kafkaHost}:{kafkaPort}";
+            var producerSettings = ProducerSettings<Null, string>.Create(sys, null, null)
+                .WithBootstrapServers(bootstrapServer);
+
+            ProducerActor = Source.ActorRef<Tuple<ISpanContext, string>>(128, OverflowStrategy.DropNew)
+                .Select(tuple =>
+                {
+                    var (spanContext, message) = tuple;
+                    var record = new ProducerRecord<Null, string>(Topic, null, message).WithTracing(sys, spanContext);
```

Comment:
The record is "enriched" with trace span context using the producer extension `WithTracing`